lib/card/set_pattern.rb

```diff
@@ -132,4 +138,23 @@ def rule_set_key
     end
   
   end
+  class TypeSet < SetPattern
+    def initialize card
+      super
+      if mod_key = lookup_inherited_key(card)
+        @inherited_key = mod_key
+      end
+    end
+
+    def lookup_inherited_key card
+      default_rule = card.rule_card(:default) and
+        type_code = default_rule.type_code and
+        default_rule.cardname.size > 2 and
+        default_rule.left.right.codename == self.class.pattern_code
```

Comment:
This tests that the *default rule is a *type set, but I wonder if that is what we want.  Are there any hazards of accepting any *default rule found here?  Are there any benefits?  Most cases, the *default value is saved (after editing) along with it's type.  It's that when the rule was for the type, we don't want to change the type that caused it to match the rule in the first place.

Maybe we just need more tests for any use cases that we think could be important.  If they work, great, if not we probably can fix it here.